src/lib/Output/Generator/Xml/FieldTypeHashGenerator.php

```diff
@@ -4,272 +4,64 @@
-        ?LoggerInterface $logger = null,
-        bool $strictMode = false
-    ) {
-        $this->normalizer = $normalizer;
-        $this->logger = $logger ?? new NullLogger();
-        $this->strictMode = $strictMode;
-    }
-
-    /**
-     * Generates the field type value $hashValue into the $writer creating an
-     * element with $hashElementName as its parent.
-     *
-     * @param \XmlWriter $writer
-     * @param string $hashElementName
-     * @param mixed $hashValue
-     */
-    public function generateHashValue(\XMLWriter $writer, $hashElementName, $hashValue)
-    {
-        $this->generateValue($writer, $hashValue, null, $hashElementName);
-    }
-
-    /**
-     * Generates $value into a serialized representation.
-     *
-     * @param \XmlWriter $writer
-     * @param mixed $value
-     * @param string|null $key
-     * @param string $elementName
-     */
-    protected function generateValue(\XmlWriter $writer, $value, $key = null, $elementName = 'value')
+    protected function generateValue(JsonObject|ArrayObject|ArrayList $parent, mixed $value): mixed
     {
         if ($value === null) {
-            $this->generateNullValue($writer, $key, $elementName);
+            return null;
         } elseif (is_bool($value)) {
-            $this->generateBooleanValue($writer, $value, $key, $elementName);
-        } elseif (is_int($value)) {
-            $this->generateIntegerValue($writer, $value, $key, $elementName);
+            return $value ? 'true' : 'false';
         } elseif (is_float($value)) {
-            $this->generateFloatValue($writer, $value, $key, $elementName);
-        } elseif (is_string($value)) {
-            $this->generateStringValue($writer, $value, $key, $elementName);
+            return sprintf('%F', $value);
         } elseif (is_array($value)) {
-            $this->generateArrayValue($writer, $value, $key, $elementName);
+            return $this->generateArrayValue($parent, $value);
         } elseif (is_object($value)) {
-            $this->generateObjectValue($value, $writer, $key, $elementName);
+            return $this->generateObjectValue($parent, $value);
         } else {
-            throw new \Exception('Invalid type in Field value hash: ' . get_debug_type($value));
+            return $value;
         }
     }
 
-    /**
-     * Generates an array value from $value.
-     *
-     * @param \XmlWriter $writer
-     * @param array $value
-     * @param string|null $key
-     * @param string $elementName
-     */
-    protected function generateArrayValue(\XmlWriter $writer, $value, $key, $elementName = 'value')
-    {
-        if ($this->isNumericArray($value)) {
-            $this->generateListArray($writer, $value, $key, $elementName);
-        } else {
-            $this->generateHashArray($writer, $value, $key, $elementName);
+    protected function generateListArray(
+        JsonObject|ArrayObject|ArrayList $parent,
+        array $listArray,
+    ): JsonObject|ArrayObject|ArrayList {
```

Comment:
You are allowed to narrow down the implementation return types (since some 7.3'ish PHP?)

```suggestion
    ): JsonObject {
```